src/Workspaces/VisualBasic/Portable/Editing/VisualBasicImportAdder.vb

```diff
@@ -142,72 +125,130 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Editing
-                '
-                '      class C : public sub M()
-                '
-                ' This Is legal code where the lambda body Is calling the instance method.  However, if we introduce a
-                ' using that brings in an extension method 'M' on 'int', then the above will become ambiguous.  This is
-                ' because lambda binding will try each interpretation separately And eliminate the ones that fail.
-                ' Adding the import will make the int form succeed, causing ambiguity.
-                '
-                ' To deal with that, we keep track of if we're in a lambda, and we conservatively assume that a method
-                ' access (even to a non-extension method) could conflict with an extension method brought in.
-
-                Dim previousInAnonymousMethod = _inAnonymousMethod
-                _inAnonymousMethod = True
-                MyBase.VisitMultiLineLambdaExpression(node)
-                _inAnonymousMethod = previousInAnonymousMethod
-            End Sub
+            Public Async Function AddPotentiallyConflictingImportsAsync(container As SyntaxNode, conflicts As HashSet(Of INamespaceSymbol), cancellationToken As CancellationToken) As Task
+                Dim nodes = ArrayBuilder(Of SyntaxNode).GetInstance()
+                Dim containsAnonymousMethods = False
+
+                CollectInfoFromContainer(container, nodes, containsAnonymousMethods)
 
-            Public Overrides Sub VisitSingleLineLambdaExpression(node As SingleLineLambdaExpressionSyntax)
-                Dim previousInAnonymousMethod = _inAnonymousMethod
-                _inAnonymousMethod = True
-                MyBase.VisitSingleLineLambdaExpression(node)
-                _inAnonymousMethod = previousInAnonymousMethod
+                Dim produceItems =
+                    Function(node As SyntaxNode, onItemsFound As Action(Of INamespaceSymbol), conflictFinder As ConflictFinder, cancellationToken1 As CancellationToken) As Task
+                        If TypeOf node Is SimpleNameSyntax Then
+                            Me.ProduceConflicts(TryCast(node, SimpleNameSyntax), onItemsFound, cancellationToken)
+                        ElseIf TypeOf node Is MemberAccessExpressionSyntax Then
+                            Me.ProduceConflicts(TryCast(node, MemberAccessExpressionSyntax), containsAnonymousMethods, onItemsFound, cancellationToken)
+                        Else
+                            Throw ExceptionUtilities.Unreachable()
+                        End If
+                        Return Task.CompletedTask
+                    End Function
+
+                Dim items = Await ProducerConsumer(Of INamespaceSymbol).RunParallelAsync(
+                    source:=nodes,
+                    produceItems:=produceItems,
+                    args:=Me,
+                    cancellationToken).ConfigureAwait(False)
+
+                For Each conflict In items
+                    conflicts.Add(conflict)
```

Comment:
conflicts.AddRange(items)

Reply:
Yeah, probably easier that way. Just was in the habit of not doing that in the case where it would box an enumerator.